astroid/inference.py

```diff
@@ -72,11 +72,11 @@ def infer_end(self, context=None):
     yield self
 
 
-nodes.Module._infer = infer_end
-nodes.ClassDef._infer = infer_end
-nodes.Lambda._infer = infer_end
-nodes.Const._infer = infer_end
-nodes.Slice._infer = infer_end
+nodes.Module._infer = infer_end  # type: ignore # See https://github.com/python/mypy/issues/2427
```

Comment:
```suggestion
nodes.Module._infer = infer_end  # type: ignore[assignment]  # See https://github.com/python/mypy/issues/2427
```
It's better if we only ignore the specific mypy error.

I haven't gotten around to it, but we should consider adding the `--show-error-codes` flag to `mypy` in both astroid and pylint. Maybe even update our existing `type: ignore` statements.

Reply:
> Hmm 🤔 Tbh. this looks rather ugly, but it seems like it's the only way at the moment (aside from `setattr`).

Would you prefer using `setattr`? I opted for `ignore` as this seems a "bug" in `mypy` and I thought this should thus be fixed by telling `mypy` to ignore this rather than to change our own code.

Reply:
> Would you prefer using `setattr`?

Not really. `type: ignore[assignment]` still allows for other type checks, `setattr` not (afaik).
Maybe it's enough to link the issue ones for the complete file though. (above `nodes.Module._infer = ...)